lib/elixir/src/elixir_expand.erl

```diff
@@ -367,8 +367,7 @@ expand({Name, Meta, Kind}, S, E) when is_atom(Name), is_atom(Kind) ->
 
         %% TODO: Remove this clause on v2.0
         _ when Error == warn ->
-          elixir_errors:form_warn(Meta, E, ?MODULE, {unknown_variable, Name}),
-          expand({Name, [{if_undefined, warn} | Meta], []}, S, E);
+          expand({Name, [{var_as_call, Kind} | Meta], []}, S, E);
```

Comment:
I think we are supposed to guarantee that `if_undefined` is always available so it is easy to check whenever there is a var as call (which also happens on `{if_undefined, apply}`). So can we add this functionality and keep the current meta?

Reply:
Done in [d415af5](https://github.com/elixir-lang/elixir/pull/12272/commits/d415af5ea1a1a285fb1de7e165118893c781ff7d), sorry for removing it without confirmation 🙇